src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -84,13 +75,10 @@ put_config(server::request_t rq, server::reply_t rp) {
       rq.req->content.data(), put_config_handler<>{});
     rq.req.reset();
 
-    auto res = rq.service().schema_store().set_compatibility(config.compat);
-    if (res.has_error()) {
-        rp.rep = make_errored_body(res.error());
-        co_return rp;
-    }
+    auto res
+      = rq.service().schema_store().set_compatibility(config.compat).value();
 
-    if (res.value()) {
+    if (res) {
```

Comment:
> although I guess I am curious here. why would we need to check res if the previous statement would throw when val isn't available? I'm probably not up-to-date on the fine details on result etc...
> 
> edit: mm, unless res itself is another layer that needs to be checked. something like `result<optional>`

In this case it's whether setting the config made a change (which then needs to be serialised to the schema topic), otherwise it's a noop.